routes/userRoutes.js

```diff
@@ -27,14 +27,59 @@ routes.get('/', async (req, res, next) => {
  * @queryString : none,
 */
 routes.get('/:id', async (req, res, next) => {
-  const { id } = req.params;
   try {
-    const user = await models.User.where({ id });
-    if (user) {
-      res.status(200).json(user);
-    } else {
-      res.status(404).json(error.getUserById);
-    }
+    const user = await models.User.findById(req.params.id).exec();
+    res.send(user);
+  } catch (error) {
+    next(error);
+  }
+});
+
+/**
+ * [POST] Creates a new user
+ * @params : none,
+ * @body : none,
+ * @queryString : none,
+ */
+routes.post('/', async (req, res, next) => {
+  const user = req.body;
+  const newUser = models.User(user);
+  try {
+    const result = await newUser.save();
+    res.status(201).json(result);
+  } catch(error) {
+    next(error);
+  }
+});
+
+/**
+ * [PUT] Updates a user by id
+ * @params : user id (integer),
+ * @body : req.body,
+ * @queryString : none,
+ */
+routes.put('/:id', async (req, res, next) => {
+  try {
+    const user = await models.User.findById(req.params.id).exec();
+    user.set(req.body);
```

Comment:
Please add following steps:
- add a check if user with `/:id` exists, if not return error message
- if the user is changing his password, it should be hashed before saving it to DB
- before sending the result object back in response, please remove the password property from it